synapseclient/table.py

```diff
@@ -777,6 +821,51 @@ def __init__(self, values, rowId=None, versionNumber=None, etag=None):
             self.etag = etag
 
 
+class PartialRow(DictObject):
+    """
+    A partial update for a single row.
+    Note "values" is a :py:class:`dict` where:
+        - key is the columnId to the desired row
+        - value is the new desired value for that column
+    The easiest way to know which columnIds to use is to check the `headers` of your most recent query:"
+        query_result = syn.TableQuery("SELECT * FROM syn123")
+        name_to_col_id = {col.name:col.id for col in query_results.headers}
+
```

Comment:
This seems pretty annoying for the user to deal with.  Why not pass in a query (i.e. TableQueryResult) instead of a name_to_col_id?